gno.land/cmd/gnoland/secrets.go

```diff
@@ -72,3 +76,84 @@ func constructSecretsPath(nodeDir string) string {
 		config.DefaultSecretsDir,
 	)
 }
+
+// printKeyValue searches and prints the given key value in JSON
+func printKeyValue[T *secrets | *config.Config](
+	input T,
+	raw bool,
+	io commands.IO,
+	key ...string,
+) error {
+	// prepareOutput prepares the JSON output, taking into account raw mode
+	prepareOutput := func(input any) (string, error) {
+		encoded, err := json.MarshalIndent(input, "", "    ")
+		if err != nil {
+			return "", fmt.Errorf("unable to marshal JSON, %w", err)
+		}
+
+		output := string(encoded)
+
+		if raw && len(encoded) > 0 && encoded[0] == '"' {
+			output = strings.ReplaceAll(output, "\"", "")
+		}
+
+		return output, nil
+	}
+
+	if len(key) == 0 {
+		// Print the entire input
+		output, err := prepareOutput(input)
+		if err != nil {
+			return err
+		}
+
+		io.Println(output)
+
+		return nil
+	}
+
+	// Get the value using reflect
+	secretValue := reflect.ValueOf(input).Elem()
+
+	// Get the value path, with sections separated out by a period
+	field, err := getFieldAtPath(secretValue, strings.Split(key[0], "."))
+	if err != nil {
+		return err
+	}
+
+	output, err := prepareOutput(field.Interface())
+	if err != nil {
+		return err
+	}
+
+	io.Println(output)
+
+	return nil
+}
+
+type (
+	secrets struct {
+		ValidatorKeyInfo   *validatorKeyInfo   `json:"validator_key_info,omitempty" toml:"validator_key_info"`
+		ValidatorStateInfo *validatorStateInfo `json:"validator_state_info,omitempty" toml:"validator_state_info"`
+		NodeIDInfo         *nodeIDInfo         `json:"node_id_info,omitempty" toml:"node_id_info"`
+	}
+
+	validatorKeyInfo struct {
+		Address string `json:"address" toml:"address"`
+		PubKey  string `json:"pub_key" toml:"pub_key"`
+	}
+
+	validatorStateInfo struct {
+		Height int64 `json:"height" toml:"height"`
+		Round  int   `json:"round" toml:"round"`
+		Step   int8  `json:"step" toml:"step"`
+
+		Signature []byte `json:"signature,omitempty" toml:"signature,omitempty"`
+		SignBytes []byte `json:"sign_bytes,omitempty" toml:"sign_bytes,omitempty"`
+	}
+
+	nodeIDInfo struct {
+		ID         string `json:"id" json:"id"`
+		P2PAddress string `json:"p2p_address" toml:"p2p_address"`
+	}
```

Comment:
Can you mark these and the corresponding structs in the config with a comment to keep them in sync? (Like with `privval.FilePVKey`)

There likely won't be anything added, but we can attempt to avoid inconsistencies if it is done by someone in the future.

Reply:
I've added comments to base + these structs:


252cad0

Reply:
Mhh, what I'm saying here is not to add a toml comment but to add a code comment, like the following.

```
// NOTE: keep in sync with gno.land/cmd/gnoland/secrets.gno
```

In the places where the secrets structs are, and vice versa.

This is a safeguard if anyone does ever modify those structs. Otherwise it's one of the things that are too easy to miss.

Reply:
Ah, I completely misunderstood then :)

@gfanton will thank me for #2399

Added notice:
[9c364cd](https://github.com/gnolang/gno/pull/2393/commits/9c364cd27ca935e380096a048931f141ac0f14fb)